volume/core.go

```diff
@@ -154,11 +150,24 @@ func (c *Core) CreateVolume(id types.VolumeID) error {
 		if err := dv.Create(ctx, v, s); err != nil {
 			return err
 		}
+
+		if err := store.MetaPut(v); err != nil {
+			return err
+		}
 	}
 
 	if f, ok := dv.(driver.Formator); ok {
-		return f.Format(ctx, v, s)
+		if err := f.Format(ctx, v, s); err != nil {
```

Comment:
I suggest that we could use the rule `return fast` to reduce idents and increase readability. So the code style would be:
```
f, ok := dv.(driver.Formator); 
if !ok {
    return nil
}

err = f.Format(ctx, v, s)
if err == nil {
    return nil
}
logrus.Errorf("failed to format new volume: %s, err: %v", v.Name, err)
logrus.Warnf("rollback create volume, start to cleanup new volume: %s", v.Name)
if err := c.RemoveVolume(id); err != nil {
	logrus.Errorf("failed to rollback create volume, cleanup new volume: %s, err: %v", v.Name, err)
    return err
}
return nil
```

Reply:
ok